examples/BCDR/BackupFromStorageQueue.cs

```diff
@@ -0,0 +1,172 @@
+/*
+ * 
+ * 
+	1. We set up Event Grid events emitted by Azure App Configuration to be routed to an Azure Storage queue. Note that event order is not guaranteed by Event grid or storage queue.  
+	2. We create a timer triggered Azure function to read contents of storage queue every <10> minutes. In this tutorial, we are using the same storage account for this Azure function and Storage Queue. 
+	3. Azure function checks if there are any messages from event grid in the queue. Any other messages in the queue in invalid format will be discarded.
+	4. If there are any messages in the queue, retrieve a batch of maximum 32 messages at a time for processing.
+	5. For all retrieved messages, extract the key+label from event message and store them in a hash set. We need to keep track of any unique key+label that was modified or deleted.
+	6. Fetch all settings from primary store. 
+	7. Update only those settings in secondary store which have a corresponding event in the storage queue.
+	8. Delete all settings that were present in storage queue but not in primary store.
+    9. Delete all messages that we received in this batch from the storage queue.
+    10. Repeat from step 4 if there are any messages remaining in storage queue.
+ *
+ *
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Azure.Storage.Queues;
+using Azure;
+using Azure.Data.AppConfiguration;
+using Azure.Identity;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+using Azure.Storage.Queues.Models;
+
+namespace BackupFromStorage
+{
+    public static class BackupFromStorageQueue
+    {
+        private static HashSet<KeyLabel> ExtractKeyLabelsFromEvents(QueueMessage[] messages, ILogger log)
+        {
+            HashSet<KeyLabel> updatedKeyLabels = new HashSet<KeyLabel>(new KeyLabelComparer());
+            foreach (QueueMessage message in messages)
+            {
+                try
+                {
+                    // Event grid will encode events in Base64 format before publishing to storage queue
+                    // Decode the message before parsing it
+                    string decodedMessage = Encoding.UTF8.GetString(Convert.FromBase64String(message.MessageText));
+                    try
+                    {
+                        JObject eventObject = JObject.Parse(decodedMessage);
+                        string topic = eventObject.Property("topic")?.Value.ToString();
+                        if (topic != null && topic == Environment.GetEnvironmentVariable("PrimaryStoreResourceId"))
```

Comment:
Defining `PrimaryStoreResourceId` is kind of an unnecessary burden to users. I guess if the message has data field with key and label properties, it should be good enough.

Reply:
Removed the check for PrimaryStoreResourceId